// rasajs/lib/utility/urls.js
/**
 * 
 * @param {string} baseUrl - base URL of the Rasa Rerver
 * @param {string} action - whitch action to execute 
 * @returns full url of the action
 */
function Urls(baseUrl,action) {
    const url = {
        parse: baseUrl + '/model/parse',
        sendmessage: baseUrl + '/webhooks/rest/webhook',
        api: baseUrl + '/api/auth',
        appendEventTracker: baseUrl + '/conversations/:conversation_id/tracker/events?include_events=AFTER_RESTART&output_channel=slack&execute_side_effects=false',
        replaceEventTracker: baseUrl + '/conversations/:conversation_id/tracker/events?include_events=AFTER_RESTART',
        retriveTracker: baseUrl + '/conversations/:conversation_id/tracker?include_events=AFTER_RESTART&until=1559744410',
        retriveStory: baseUrl + '/conversations/:conversation_id/story?until=1559744410&all_sessions=false',
        actionConversation: baseUrl + '/conversations/:conversation_id/execute?include_events=AFTER_RESTART&output_channel=slack',
        injectIntent: baseUrl + '/conversations/:conversation_id/trigger_intent?include_events=AFTER_RESTART&output_channel=slack',
        evaluateStories: baseUrl + '/model/test/stories?e2e=false',
        replaceModel: baseUrl + '/model',
        domain: baseUrl + '/domain',
        version: baseUrl + '/version',
        status: baseUrl + '/status'
    };
    return url[action];
}
export {Urls as default};
